github-workitem/src/main/java/org/jbpm/process/workitem/github/MergePullRequestWorkitemHandler.java

```diff
@@ -80,31 +79,25 @@ public void executeWorkItem(WorkItem workItem,
 
             MergeStatus mergeStatus;
 
-            if (StringUtils.isNotEmpty(repoOwner) && StringUtils.isNotEmpty(repoName) && StringUtils.isNumeric(pullRequestNum)) {
-
-                PullRequestService pullRequestService = auth.getPullRequestService(this.userName,
-                                                                                   this.password);
-
-                RepositoryId repositoryId = new RepositoryId(repoOwner,
-                                                             repoName);
-                if (pullRequestService.getPullRequest(repositoryId,
-                                                      Integer.parseInt(pullRequestNum)).isMergeable()) {
-                    mergeStatus = pullRequestService.merge(repositoryId,
-                                                           Integer.parseInt(pullRequestNum),
-                                                           commitMessage);
-                    results.put(RESULTS_VALUE,
-                                mergeStatus.isMerged());
-                } else {
-                    results.put(RESULTS_VALUE,
-                                false);
-                }
-
-                workItemManager.completeWorkItem(workItem.getId(),
-                                                 results);
+            PullRequestService pullRequestService = auth.getPullRequestService(this.userName,
+                                                                               this.password);
+
+            RepositoryId repositoryId = new RepositoryId(repoOwner,
+                                                         repoName);
+            if (pullRequestService.getPullRequest(repositoryId,
+                                                  Integer.parseInt(pullRequestNum)).isMergeable()) {
+                mergeStatus = pullRequestService.merge(repositoryId,
+                                                       Integer.parseInt(pullRequestNum),
+                                                       commitMessage);
+                results.put(RESULTS_VALUE,
+                            mergeStatus.isMerged());
             } else {
-                logger.error("Missing repository and pull request info.");
-                throw new IllegalArgumentException("Missing repository and pull request info.");
```

Comment:
why we are changing this to not throw exception anymore?

Reply:
It's no longer needed any more since we added call to

RequiredParameterValidator.validate(this.getClass(),workItem);

and 

https://github.com/kiegroup/jbpm/blob/master/jbpm-workitems/jbpm-workitems-core/src/main/java/org/jbpm/process/workitem/core/util/RequiredParameterValidator.java#L41
will throw the exception now for required params. The checks for existing params no longer needed.

Reply:
True, that’s not needed from validation point of view but shouldn’t it be considered as error in case pull request is not mergable?

Reply:
Good point - I'll fix. Thanks!